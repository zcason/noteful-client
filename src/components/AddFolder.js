import React, { Component } from 'react';
import SidebarSection from './sections/SidebarSection';
import DefaultContext from './context/DefaultContext';
import { withRouter } from 'react-router-dom';
import ErrorBoundary from '../errors/ErrorBoundary';
import PropTypes from 'prop-types';

class AddFolder extends Component {
    static contextType = DefaultContext;
    handleOnSumbit = (form) => {
        const f = new FormData(form);
        const name = f.get("folderName");
        if ( typeof name !== "string" || name.length <= 0 ) return;
        this.addFolder({name:name});
    }
    addFolder = (data) => {
        fetch(`${this.context.url}/folders/`, {
            method: 'POST',
            headers: {
              'content-type': 'application/json'
            },
            body: JSON.stringify(data)
        })
        .then( r => {
            this.context.updateStore();
            this.props.history.push('/');
        } )
        .catch( e => {
            throw new Error("Error creating folder");
        } );
    }
    render() {
        return (
            <div className="App">
                <ErrorBoundary message="Sidebar Section Error">
                    <SidebarSection 
                        store={this.props.store}
                        history={this.props.history}
                    />
                </ErrorBoundary>
                <ErrorBoundary message="Main Section Error">
                    <main className="section--main">
                        <form
                            className="add--form"
                            onSubmit={(e)=>{
                                e.preventDefault();
                                this.handleOnSumbit(e.target);
                            }}
                        >   
                            <div className="add--form--field">
                                <label htmlFor="folderName">Folder name:</label>
                                <input type="text" id="folderName" name="folderName" required />
                            </div>
                            <div className="add--form--field">
                                <button type="submit">Submit</button>
                            </div>
                        </form>
                    </main>
                </ErrorBoundary>
            </div>
        )
    }
}

AddFolder.propTypes = {
    store: PropTypes.object.isRequired
}

export default withRouter(AddFolder);