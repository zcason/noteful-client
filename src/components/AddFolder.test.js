import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import store from '../dummy-store';
import AddFolder from './AddFolder';

describe(`AddFolder component`, () => {
    const props = {
      store: store
    }
  
    it('renders AddFolder by default', () => {
      const wrapper = shallow(<AddFolder />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('renders form from store when in props', () => {
      const component = shallow(<AddFolder {...props} />)
        .find('form')
      expect(toJson(component)).toMatchSnapshot()
    })
  })
  