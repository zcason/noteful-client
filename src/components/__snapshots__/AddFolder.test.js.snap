// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFolder component renders AddFolder by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`AddFolder component renders form from store when in props 1`] = `null`;
